receiver/postgresqlreceiver/integration_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration
+// +build integration
+
+package postgresqlreceiver
+
+import (
+	"context"
+	"net"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/scrapertest"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/scrapertest/golden"
+)
+
+type configFunc func(hostname string) *Config
+
+type testCase struct {
+	name         string
+	cfg          configFunc
+	expectedFile string
+}
+
+func TestPostgreSQLIntegration(t *testing.T) {
+	testCases := []testCase{
+		testCase{
+			name: "single_db",
+			cfg: func(hostname string) *Config {
+				f := NewFactory()
+				cfg := f.CreateDefaultConfig().(*Config)
+				cfg.Endpoint = net.JoinHostPort(hostname, "5432")
+				cfg.Databases = []string{"otel"}
+				cfg.Username = "otel"
+				cfg.Password = "otel"
+				cfg.Insecure = true
+				return cfg
+			},
+			expectedFile: filepath.Join("testdata", "integration", "expected_single_db.json"),
+		},
+		testCase{
+			name: "multi_db",
+			cfg: func(hostname string) *Config {
+				f := NewFactory()
+				cfg := f.CreateDefaultConfig().(*Config)
+				cfg.Endpoint = net.JoinHostPort(hostname, "5432")
+				cfg.Databases = []string{"otel", "otel2"}
+				cfg.Username = "otel"
+				cfg.Password = "otel"
+				cfg.Insecure = true
+				return cfg
+			},
+			expectedFile: filepath.Join("testdata", "integration", "expected_multi_db.json"),
+		},
+		testCase{
+			name: "all_db",
+			cfg: func(hostname string) *Config {
+				f := NewFactory()
+				cfg := f.CreateDefaultConfig().(*Config)
+				cfg.Endpoint = net.JoinHostPort(hostname, "5432")
+				cfg.Databases = []string{}
+				cfg.Username = "otel"
+				cfg.Password = "otel"
+				cfg.Insecure = true
+				return cfg
+			},
+			expectedFile: filepath.Join("testdata", "integration", "expected_all_db.json"),
```

Comment:
We always prepend `testdata/integration`, maybe we should do that in the loop and make `expectedFile` be relative to `testdata/integration`?